rocAL/include/pipeline/argument.h

```diff
@@ -0,0 +1,261 @@
+    std::string type_name;                ///< Denotes the data type of the argument
+    std::string sub_type_name;            ///< Denotes the name of the enum/ type of parameter
+    bool is_vector = false;               ///< True if the argument contains vector data
+    bool is_parameter = false;            ///< True if the argument is a parameter object
+    bool is_null_ptr = false;             ///< True if the argument represents a null pointer
+    std::vector<std::any> values;         ///< Storage for argument values
+    pParam param;                         ///< Parameter stored for parameter-type arguments
+
+    // Constructors
+
+    /**
+     * @brief Unified template constructor for all data types
+     * @tparam T The type of the value being stored
+     * @param name The name of the argument
+     * @param val The value to store
+     * @throws std::runtime_error if the type is unknown or unsupported
+     */
+    template <typename T>
+    explicit Argument(std::string name, T&& val) : arg_name(std::move(name)) {
+
+        if constexpr (std::is_enum_v<std::decay_t<T>>) {
+            constructFromEnum(std::forward<T>(val));
+        } else if constexpr (is_vector_type_v<std::decay_t<T>>) {
+            constructFromVector(std::forward<T>(val));
+        } else if constexpr (is_shared_ptr_v<std::decay_t<T>>) {
+            constructFromSharedPtr(std::forward<T>(val));
+        } else if constexpr (is_string_map_v<std::decay_t<T>>) {
+            constructFromMap(std::forward<T>(val));
+        } else if constexpr (is_param_type_v<std::decay_t<T>>) {
+            constructFromParam(std::forward<T>(val));
+        } else {
+            constructFromBasicType(std::forward<T>(val));
+        }
+    }
+
+
+private:
+    /**
+     * @brief Helper method to get type name from registry or built-in types
+     * @tparam T The type to get the name for
+     * @return The string representation of the type name
+     */
+    template<typename T>
+    std::string getTypeName() const {
+        using DecayedType = std::decay_t<T>;
+        
+        if constexpr (std::is_enum_v<DecayedType>) {
+            // For enum types, check the registry first
+            std::string enum_name = EnumRegistry::getInstance().getEnumName<DecayedType>();
+            return enum_name.empty() ? "unknown_enum" : enum_name;
+        } else {
+            // Use the type name resolution from argument_types.h
+            return std::string(get_type_name<DecayedType>());
+        }
+    }
+
+    /**
+     * @brief Constructs argument from enum type
+     * @tparam T The enum type
+     * @param val The enum value
+     */
+    template<typename T>
+    void constructFromEnum(T&& val) {
+        static_assert(std::is_enum_v<std::decay_t<T>>, "T must be an enum type");
+        
+        type_name = "enum";
+        sub_type_name = getTypeName<T>();
+        
+        if (sub_type_name != "unknown_enum") {
+            values.push_back(static_cast<int>(val));
```

Comment:
Blocking: `static_cast<int>(val)` assumes every enum fits in a 32-bit signed int. Because the registry now allows registering enums with wider or unsigned underlying types (e.g. `enum class Large : uint64_t`), serializing them through this path will truncate or wrap the value. Please store the value as `std::underlying_type_t<std::decay_t<T>>` (and do the same for the vector case) instead of forcing everything through `int`.

Reply:
Ignoring this comment, as all the enums are generally considered int